File: app_be/user-service/src/main/java/com/medibook/user/dto/MedicalServiceDto.java
package com.medibook.user.dto;

import lombok.*;

import java.util.UUID;

/**
 * DTO cho Medical Service response
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MedicalServiceDto {
    private UUID id;
    private String name;
    private String description;
    private Double price;
    private Integer durationMinutes;
    private String category;
    private Boolean isActive;
    private String imageUrl;
}

File: app_admin/lib/features/services/data/dto/medical_service_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'medical_service_dto.freezed.dart';
part 'medical_service_dto.g.dart';

@freezed
class MedicalServiceDto with _$MedicalServiceDto {
  const factory MedicalServiceDto({
    required String id,
    required String name,
    required String? description,
    required double price,
    required int durationMinutes,
    required String category,
    @Default(true) bool isActive,
    required String? imageUrl,
  }) = _MedicalServiceDto;

  factory MedicalServiceDto.fromJson(Map<String, dynamic> json) =>
      _$MedicalServiceDtoFromJson(json);
}
